File: ch1/string_rotation/cc/stringRotation.cc
#include <iostream>
#include <string>
#include <iomanip>
using namespace std;

bool stringRotation(string a, string b){
    for(int i = 0; i < b.length(); i++){
        if(a == b){
            return true;
        }
        b = b.substr(1,b.length()) + b[0];
    }
    return false;
}

int main(int argc, char** argv){
    if(argc != 3){
        printf("Usage: $ ./stringRotation word1 word2\n");
        return 1;
    }
    cout << boolalpha << stringRotation(argv[1],argv[2]) << endl;
    return 0;
}

File: ch1/string_rotation/python/stringRotation.py
import sys

def stringRotation(word1, word2):
    for i in range(0, len(word2)):
        if word1 == word2:
            return True
        else:
            word2 = word2[1:len(word2)] + word2[0]
    return False

def main():
    if len(sys.argv) < 1:
        print("Usage: $ python stringRotation.py word1 word2")
    else:
        print(stringRotation(sys.argv[1], sys.argv[2]))

if __name__ == "__main__":
    main()
